libr/debug/p/native/linux/linux_debug.c

```diff
@@ -517,3 +521,1439 @@ RList *linux_desc_list (int pid) {
+			break;
+		case 'p':
+			flags |= P_MEM;
+			break;
+		case 's':
+			flags |= S_MEM;
+			break;
+		case '-':
+			break;
+		}
+	}
+	
+	*fl_perms = flags;
+	
+	if (((flags & P_MEM) && (flags & S_MEM)) || (!(flags & R_MEM) && !(flags & W_MEM))) {
+		eprintf ("setting WRG_PERM\n");
+		*fl_perms = WRG_PERM;
+	}
+}
+
+static void get_map_offset(char *pstr, ut64 *offset) {
+	char *pp;
+	
+	pp = pstr;
+	*offset = strtoul (pp, &pp, 16);
+}
+
+
+static void get_map_name(char *pstr, char **name) {
+	*name = strdup (pstr);
+}
+
+static bool has_map_deleted_part(char *name) {
+	char deleted_str[] = "(deleted)";
+	int len_suffx;
+	int len_name;
+	int ret;
+	
+	len_name = strlen (name);
+	len_suffx = strlen (deleted_str);
+	
+	ret = strncmp (name + len_name - len_suffx, deleted_str, len_suffx);
+	
+	return ret ? false : true;
+}
+	
+
+static bool get_anonymous_value(char *keyw) {
+	keyw = strchr (keyw, ' ');
+
+	while (isspace (*keyw))
+		keyw++;
+
+	return *keyw != '0';
+}
+
+static bool has_map_anonymous_content(char *buff_smaps, ut64 start_addr, ut64 end_addr) {
+	char identity[80];
+	char *str;
+	char *p;
+	char *pp;
+	char *extern_tok;
+	char *keyw;
+	bool is_anonymous;
+
+	snprintf (identity, sizeof (identity), "%08"PFMT64x"-%08"PFMT64x"", start_addr, end_addr);
+	str = strdup (buff_smaps);
+
+	for (p = strtok_r (str, "\n", &extern_tok) ; p ; p = strtok_r (NULL, "\n", &extern_tok)) {
+		if (strstr (p, identity) != NULL) { 
+			for ( pp = strtok_r (NULL, "\n", &extern_tok); pp ; pp = strtok_r (NULL, "\n", &extern_tok)) {
+				if ((keyw = strstr (pp, "Anonymous:")) != NULL || (keyw = strstr (pp, "AnonHugePages:")) != NULL) {
+					is_anonymous = get_anonymous_value (keyw);
+					free (str);
+					return is_anonymous;
+				}
+			}
+		}
+	}
+	
+	free (str);
+	return 0;
+}
+
+static bool is_a_kernel_mapping(char *map_name) {
+	bool ret;
+
+	if (strcmp (map_name, "[vsyscall]") == 0 ||
+		strcmp (map_name, "[vvar]") == 0 ||
+		strcmp (map_name, "[vdso]") == 0) {
+		ret = true;
+	} else {
+		ret = false;
+	}
+
+	return ret;
+}
+
+static bool dump_this_map(char *buff_smaps, ut64 start_addr, ut64 end_addr, bool file_backed, bool anonymous, ut8 perms, ut8 filter_flags) {
+	char identity[80];
+	char *aux;
+	char *p;
+	char *pp;
+	char *ppp;
+	char *extern_tok;
+	char *flags_str;
+	bool found;
+	unsigned char vmflags;
+
+	/* if the map doesn't have r/w quit right here */
+	if (perms & WRG_PERM) {
+		eprintf("[dump_this_map] wrong perm detected on %"PFMT64x"-%"PFMT64x"\n", start_addr, end_addr);
+		return false;
+	}
+
+	eprintf ("[dump_this_map] %"PFMT64x"-%"PFMT64x": file: %d - anonymous - %d - flags: 0%x\n", start_addr, end_addr, file_backed, anonymous, filter_flags);
+
+	snprintf (identity, sizeof(identity), "%08"PFMT64x"-%08"PFMT64x"", start_addr, end_addr);
+
+	found = 0;
+	vmflags = 0;
+	flags_str = NULL;
+	aux = strdup (buff_smaps);
+
+	for (pp = strtok_r (aux, "\n", &extern_tok) ; pp ; pp = strtok_r (NULL, "\n", &extern_tok)) {
+		if (strstr(pp, identity)) {
```

Comment:
Missing spaces. Is better to negate the condition and continue to avoid the christmas tree